test/topology/test_find_related.py

```diff
@@ -17,12 +17,13 @@
 #  You should have received a copy of the GNU General Public License
 #  along with segment-reshape-qgis-plugin. If not, see <https://www.gnu.org/licenses/>.
 
-from typing import List, Optional, Tuple
+from typing import Callable, List, Optional, Tuple
 
 import pytest
-from qgis.core import QgsFeature, QgsGeometry, QgsPoint, QgsVectorLayer
+from qgis.core import QgsFeature, QgsGeometry, QgsPoint, QgsProject, QgsVectorLayer
 
 from segment_reshape.topology import find_related
+from segment_reshape.topology.find_related import find_related_features
```

Comment:
Should we just use module imports in the future at least for our own modules (not sure about QGIS modules but I guess we could do it there also?) as it has some benefits: no merge conflicts from imports and when mocking functions mocker.patch.object can be used (IDE refactor works e.g. if module name is changed).

Also, this would be in-line with google python style guide: https://google.github.io/styleguide/pyguide.html#22-imports

Reply:
Well there are obviously benefits (also listed somewhere that some circular import are avoided if imported modules are not used on the module top level code to resolve other names, classes etc.), but some module naming changes 

For example here we have `segment_reshape.topology.find_related` with name `find_related_features`, used as is would be very verbose duplication like `segment_reshape.topology.find_related.find_related_features` or slightly shorter but duplicate still `find_related.find_related_features`, where actually it should be more like `topology.find_related_features` since we probably don't have two same public names in a submodule namespace? This would require either using init files to collect the public names if its still beneficial to split code to own modules in `topology` module namespace.

For testing, there is a semantic (and most likely functional in complex code base) difference when using `mocker.patch("other.module.that.imports.topology")` and `mocker.patch.object(topology)`: plain patch will only mock the name in the location specified, and other modules will use the original, and object patch will replace the function fully. If module `x` uses `utils.y` and `utils.z` in its code, testing requires to mock some return for `utils.y` call, but actually `utils.z` uses `y` internally also, behavior of `utils.z` changes due to patched `utils.y`, so you need to know internals of `utils` for unit testing `x`, if all of `utils` is not mocked to return certain things. Probably suboptimal for own code, even more difficult for dependencies.